docs/migration/v8-to-v9.md

```diff
@@ -0,0 +1,159 @@
+# Upgrading from 8.x to 9.x
+
+**DISCLAIMER: THIS MIGRATION GUIDE IS WORK IN PROGRESS**
+
+Version 9 of the Sentry SDK concerns itself with API cleanup and compatibility updates.
+This update contains behavioral changes that will not be caught by TypeScript or linters, so we recommend carefully reading the section on [Behavioral Changes](#2-behavior-changes).
```

Comment:
meta: I just realized that our comments/docs are a mix of american and british english (for ex: behavioral vs. behavioural) because of me. I made it Behaviour in the v8 migration docs.

We should probably mandate american english spelling to match the docs (even though `colour` is vastly superior to `color` in every way).